version: "2.1"

volumes:
  pgdata:
  walletstore:

networks:
  test_akb:
    name: test_akb
    external: true

services:
  explorerdb.akb:
    image: hyperledger/explorer-db:latest
    container_name: explorerdb.akb
    hostname: explorerdb.akb
    environment:
      - DATABASE_DATABASE=fabricexplorerakb
      - DATABASE_USERNAME=akb
      - DATABASE_PASSWORD=password
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -q -U postgres"
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - pgdata:/var/lib/postgresql/data
    networks:
      - test_akb

  explorer.akb:
    image: lucaakb/akb:akbexplorer
    container_name: explorer.akb
    hostname: explorer.akb
    environment:
      - DATABASE_HOST=explorerdb.akb
      - DATABASE_DATABASE=fabricexplorerakb
      - DATABASE_USERNAME=akb
      - DATABASE_PASSWD=password
      - LOG_LEVEL_APP=info
      - LOG_LEVEL_DB=info
      - LOG_LEVEL_CONSOLE=debug
      - LOG_CONSOLE_STDOUT=true
      - DISCOVERY_AS_LOCALHOST=false
      - PORT=8082
      - REACT_APP_API_URL=http://localhost:8000
    volumes:
      - ./config.json:/opt/explorer/app/platform/fabric/config.json
      - ./connection-profile/:/opt/explorer/app/platform/fabric/connection-profile
      - ./../organizations:/tmp/crypto
      - walletstore:/opt/explorer/wallet
    ports:
      - 8082:8082
    depends_on:
      explorerdb.akb:
        condition: service_healthy
    networks:
      - test_akb
